syntax = "proto3";

package echo;

// Echo service definition
service EchoService {
  // Simple echo method
  rpc Echo (EchoRequest) returns (EchoResponse) {}

  // Server streaming echo
  rpc ServerStreamingEcho (EchoRequest) returns (stream EchoResponse) {}
}

// Echo request message
message EchoRequest {
  string message = 1;
}

// Echo response message
message EchoResponse {
  string message = 1;
  string timestamp = 2;
}